dgp/utils/render_3d_to_2d.py

```diff
@@ -0,0 +1,422 @@
+    lidar_datum_names: Optional[list[str]], max_num_items: Optional[int]
+):
+    """Draw the distance's distributution histogram of 2d bounding boxes by associating bounding_box_3d of the same object.
+    Parameters
+    ----------
+    scenes_dataset_json: str
+        Full path to the dataset scene json.
+    ontology_name_mapper: dict
+        Map the class names from bounding_box_2d to bounding_box_3d if the class names are different.
+        eg: {'Pedestrian': 'Person','Car': 'Car'}.
+    output_dir: str
+        Path to save the histogram picture.
+    camera_datum_names: Optional[list[str]]
+        List of camera names.
+        If None, use all the cameras available in the DGP dataset.
+    lidar_datum_names: Optional[list[str]]
+        List of lidar names.
+        If None, use all the lidars available in the DGP dataset.
+    max_num_items: Optional[int]
+        If not None, then show only up to this number of items. This is useful for debugging a large dataset.
+        Default: None.
+    """
+    associated_bboxes = associate_3d_and_2d_annotations_dataset(
+        scenes_dataset_json, ontology_name_mapper, camera_datum_names, lidar_datum_names, max_num_items
+    )
+    os.makedirs(output_dir, exist_ok=True)
+    # Summarize statistics per camera per class over all scenes.
+    for camera_name in camera_datum_names:
+        for name_2d, name_3d in ontology_name_mapper.items():
+            logging.info("Summarizing class {}".format(name_2d))
+            summarize_3d_statistics_per_class(associated_bboxes[camera_name][name_2d], output_dir, camera_name, name_2d)
+
+
+def summarize_3d_statistics_per_class(
+    associated_bboxes: list[Tuple], output_dir: str, camera_name: str, class_name: str
+):
+    """Accumulate distances of the associated bounding boxes and draw the histogram.
+    Parameters
+    ----------
+    associated_bboxes: list[Tuple]
+        A list of Tuple [bounding_box_2d, bounding_box_3d, centroid_2d]
+    output_dir: str
+        Path to save the histogram picture.
+    camera_name: str
+        camera name.
+    class_name: str
+        Class name.
+    """
+    dist = []
+    for _, bbox_lidar, __ in associated_bboxes:
+        dist.append(
+            int(
+                math.sqrt((bbox_lidar.pose.tvec[0]) * (bbox_lidar.pose.tvec[0]) + (bbox_lidar.pose.tvec[1]) *
+                          (bbox_lidar.pose.tvec[1]))
+            )
+        )
+    draw_hist(dist, output_dir, xlable='Dist', title=f"dist_{camera_name}_{class_name}")
+
+
+def draw_hist(data: list, output_dir: str, xlable: str, title: str):
+    """Draw the histogram of given data.
+    Parameters
+    ----------
+    data: list
+        A list of Tuple [bounding_box_2d, bounding_box_3d, centroid_2d]
+    output_dir: str
+        Path to save the histogram picture.
+    xlable: str
+        The label name of x.
+    title: str
+        The tile of the picture.
+    """
+    data = np.array(data)
```

Comment:
data is listed as "A list of Tuple [bounding_box_2d, bounding_box_3d, centroid_2d]", do we have a numpy array of objects?